apiserver/stateauthenticator/context.go

```diff
@@ -54,11 +54,20 @@ type AccessService interface {
 	// given name.
 	UpdateLastModelLogin(ctx context.Context, name string, modelUUID coremodel.UUID) error
 
-	// ReadUserAccessForTarget returns the access that
+	// ReadUserAccessLevelForTarget returns the access that
```

Comment:
This is no longer needed. You can remove it from the access service interfaces.
```diff
diff --git a/apiserver/stateauthenticator/context.go b/apiserver/stateauthenticator/context.go
index 3ccc8636c3..bcba8c0646 100644
--- a/apiserver/stateauthenticator/context.go
+++ b/apiserver/stateauthenticator/context.go
@@ -54,12 +54,6 @@ type AccessService interface {
 	// given name.
 	UpdateLastModelLogin(ctx context.Context, name string, modelUUID coremodel.UUID) error
 
-	// ReadUserAccessLevelForTarget returns the access that
-	// the input user subject has for the input target.
-	ReadUserAccessLevelForTarget(
-		ctx context.Context, subject string, target corepermission.ID,
-	) (corepermission.Access, error)
-
 	// ReadUserAccessLevelForTargetAddingMissingUser returns the user access level for
 	// the given user on the given target. If the user is external and does not yet
 	// exist, it is created. An accesserrors.AccessNotFound error is returned if no
diff --git a/internal/worker/httpserverargs/authenticator.go b/internal/worker/httpserverargs/authenticator.go
index 2522084908..c8582b6e82 100644
--- a/internal/worker/httpserverargs/authenticator.go
+++ b/internal/worker/httpserverargs/authenticator.go
@@ -42,10 +42,6 @@ type AccessService interface {
 	// given name on the given model.
 	UpdateLastModelLogin(ctx context.Context, name string, modelUUID coremodel.UUID) error
 
-	// ReadUserAccessLevelForTarget returns the access that
-	// the input user subject has for the input target.
-	ReadUserAccessLevelForTarget(ctx context.Context, subject string, target permission.ID) (permission.Access, error)
-
 	// ReadUserAccessLevelForTargetAddingMissingUser returns the user access level for
 	// the given user on the given target. If the user is external and does not yet
 	// exist, it is created. An accesserrors.AccessNotFound error is returned if no
diff --git a/internal/worker/httpserverargs/worker.go b/internal/worker/httpserverargs/worker.go
index 96c7625701..170b109d47 100644
--- a/internal/worker/httpserverargs/worker.go
+++ b/internal/worker/httpserverargs/worker.go
@@ -171,14 +171,6 @@ func (b *managedServices) GetUserByName(ctx context.Context, name string) (coreu
 	return b.accessService.GetUserByName(b.tomb.Context(ctx), name)
 }
 
-// ReadUserAccessLevelForTarget returns the access that
-// the input user subject has for the input target.
-func (b *managedServices) ReadUserAccessLevelForTarget(
-	ctx context.Context, subject string, target permission.ID,
-) (permission.Access, error) {
-	return b.accessService.ReadUserAccessLevelForTarget(b.tomb.Context(ctx), subject, target)
-}
-
 // ReadUserAccessLevelForTargetAddingMissingUser returns the user access level for
 // the given user on the given target. If the user is external and does not yet
 // exist, it is created.
```

Reply:
Removed